sdk/state/open.go

```diff
@@ -251,8 +254,6 @@ func (c *Channel) ConfirmOpen(m OpenAgreement) (open OpenAgreement, err error) {
 		return OpenAgreement{}, fmt.Errorf("validating open agreement: %w", err)
 	}
 
-	c.startingSequence = c.initiatorEscrowAccount().SequenceNumber + 1
-
```

Comment:
❓  this was implicitly checking that the EI.SequenceNumber is correct (otherwise the signature validation fails). How does the responder know now if the proposed sequence number is correct?

or does the responder just assume it's correct, and if it's a lesser value then the formation fails and channel doesn't open?

Reply:
I initially had the same concern. However, after more thought, there's no way the responder can know for sure whether a sequence is correct or not, even if the responder was to attempt to determine the sequence themselves it might not be correct. The only thing the responder can check is if the sequence is in the past. If it's in the past – i.e. the sequence is less than the current escrow account sequence – then there's a good chance the initiator is just trying to waste the responders time. I don't think it is really worth checking for that though as there's no material impact.